src/value_generators/include/value_generators/DocumentGenerator.hpp

```diff
@@ -57,6 +62,35 @@ struct GeneratorArgs {
     ActorId actorId;
 };
 
+class Appendable {
+public:
+    virtual ~Appendable() = default;
+    virtual void append(const std::string& key, bsoncxx::builder::basic::document& builder) = 0;
+    virtual void append(bsoncxx::builder::basic::array& builder) = 0;
+};
+
+using UniqueAppendable = std::unique_ptr<Appendable>;
+
+template <class T>
+class Generator : public Appendable {
+public:
+    ~Generator() override = default;
+    virtual T evaluate() = 0;
+    void append(const std::string& key, bsoncxx::builder::basic::document& builder) override {
+        builder.append(bsoncxx::builder::basic::kvp(key, this->evaluate()));
+    }
+    void append(bsoncxx::builder::basic::array& builder) override {
+        builder.append(this->evaluate());
+    }
+};
+
+template <class T>
+using UniqueGenerator = std::unique_ptr<Generator<T>>;
+const static boost::posix_time::ptime epoch{boost::gregorian::date(1970, 1, 1)};
+
+
+UniqueGenerator<int64_t> intGenerator(const Node& node, GeneratorArgs generatorArgs);
+UniqueGenerator<double> doubleGenerator(const Node& node, GeneratorArgs generatorArgs);
```

Comment:
I only needed to expose this line, but I think I need all of the stuff above it in order to expose. This is moved out from DocumentGenerator.cpp

Reply:
I'm afraid of having some decls here and others just in the cpp without any real rationale for what lives where beyond "shrug we needed this to compile."

I say this without a real suggestion for how to fix it / I would need to fight the compiler myself on this. I think leave this as a TODO or a comment here and in the cpp noting that there's not yet any strong intuition about what's forward-declared here versus compilation-unit-defined in the cpp.

Reply:
Would a pimpl around these two things make all the other crap go away?